### DISCUSSION TITLE
Serialize a pointer array

### DISCUSSION THREAD

**Discussion by Fewnity** [nlohmann/json#3745] [Q&A] [answered]
Hello I'm trying to serialize a big array, but I can't find a solution.

`int* myArray = (int*)malloc(size * sizeof(int));`
How I can serialize this please?
(PS : I can create a normal array like `int myArray [size];` because my array is too big for the compiler)

Thanks!

**Comment by falbrechtskirchinger** [comment, marked as answer]
Assuming you have good reason to use `malloc()` in a C++ application, you can do the following:
```cpp
using nlohmann::json;
json j(std::vector<json::number_integer_t>(myArray, myArray + size));
```

Note that `json{...}` will create an array of array, so be sure to use parentheses.

**Reply by Fewnity** [reply]
Okay thank you, I can create my file but now my problem is when I'm trying to read the value like:
```
json data = json::parse(buffer);
int t = data["pixelsColor"][0]);
```
I have an error that say : type must be number, but is binary (from json.hpp line 4928), how I can convert a binary value to int? Thanks for your help

**Reply by Fewnity** [reply]
I found the solution, thanks.
